typescript/infra/src/core/deploy.ts

```diff
@@ -1,161 +1,172 @@
-import path from 'path';
-import { ethers } from 'ethers';
-import { types } from '@abacus-network/utils';
-import {
-  AbacusCore,
-  ChainName,
-  CoreContractAddresses,
-  ProxiedAddress,
-} from '@abacus-network/sdk';
-import { AbacusAppDeployer, ProxiedContract } from '@abacus-network/deploy';
 import {
-  UpgradeBeaconController,
-  XAppConnectionManager,
   InboxValidatorManager,
   InboxValidatorManager__factory,
+  Inbox__factory,
+  InterchainGasPaymaster__factory,
   OutboxValidatorManager,
   OutboxValidatorManager__factory,
-  Inbox,
+  Outbox__factory,
+  UpgradeBeaconController,
   UpgradeBeaconController__factory,
+  XAppConnectionManager,
   XAppConnectionManager__factory,
-  Outbox__factory,
-  Inbox__factory,
-  InterchainGasPaymaster__factory,
 } from '@abacus-network/core';
+import { AbacusAppDeployer, ProxiedContract } from '@abacus-network/deploy';
+import {
+  AbacusCore,
+  ChainName,
+  CoreContractAddresses,
+  Mailbox,
+} from '@abacus-network/sdk';
+import { Remotes } from '@abacus-network/sdk/dist/types';
+import { types } from '@abacus-network/utils';
+import { ethers } from 'ethers';
+import path from 'path';
 import { DeployEnvironment, RustConfig } from '../config';
-import { CoreConfig, ValidatorManagerConfig } from './types';
+import { CoreConfig } from './types';
 
 export class AbacusCoreDeployer extends AbacusAppDeployer<
-  CoreContractAddresses,
-  CoreConfig
+  CoreContractAddresses<ChainName, any>,
```

Comment:
Extending `AbacusAppDeployer` has become too complicated, we need to simplify the types here significantly, at least back to where it was before.